subsys/bluetooth/host/scan.h

```diff
@@ -5,12 +5,74 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef SUBSYS_BLUETOOTH_HOST_SCAN_H_
+#define SUBSYS_BLUETOOTH_HOST_SCAN_H_
+
+#include <stdint.h>
+
+#include <zephyr/sys/atomic.h>
+#include <zephyr/bluetooth/bluetooth.h>
+
+/**
+ * Reasons why a scanner can be running.
+ * Used as input to @ref bt_le_scan_update_and_reconfigure.
+ */
+enum scan_enabled_reason {
+	/** The application explicitly instructed the stack to scan for advertisers
+	 * using the API @ref bt_le_scan_start().
+	 * May not be used explicitly.
+	 */
+	SCAN_ENABLED_REASON_EXPLICIT_SCAN,
+
+	/**
+	 * Syncing to a periodic advertiser.
+	 */
+	SCAN_ENABLED_REASON_SYNC_SYNCING,
+
+	/**
+	 * Scanning to find devices to connect to.
+	 */
+	SCAN_ENABLED_REASON_SCAN_BEFORE_INITIATE,
+
+	/**
+	 * Special state for a NOP for @ref bt_le_scan_update_and_reconfigure to not add/remove any
+	 * reason why the scanner should be running.
+	 */
+	SCAN_UPDATE_REASON_NONE,
+	SCAN_ENABLED_REASON_NUM_FLAGS,
```

Comment:
Suggest to add `bt_le_scan` as the prefix for these to be more consistent with the API (even if it is an internal API)